core/src/epicli/data/common/ansible/playbooks/roles/helm/tasks/install-chart.yml

```diff
@@ -9,11 +9,17 @@
 
   always:
     - name: Clean up temporary Helm chart values file
+      delegate_to: localhost
       file:
         state: absent
         path: /tmp/{{ helm_chart_name }}_values.yaml
 
   block:
+    - name: Upgrade/install Helm chart
+      delegate_to: localhost
+      environment:
+        KUBECONFIG: "{{ vault_location }}/../kubeconfig"
```

Comment:
It should be on the same level, in block section

Reply:
@rpudlowski93 are you sure everything is OK here? :D The whole block:
```
    - name: Upgrade/install Helm chart
      delegate_to: localhost
      environment:
        KUBECONFIG: "{{ vault_location }}/../kubeconfig"
```
Does nothing. It's incorrect, it's unneded, it's a noop instruction. 🤗
Please look at the bigger picture here https://github.com/epiphany-platform/epiphany/blob/ea311c24991a965add4bed773ddda028bb120622/core/src/epicli/data/common/ansible/playbooks/roles/helm/tasks/install-chart.yml#L18-L21

@ar3ndt please remove it. :+1:

Reply:
I think Robert is right here. I cant really remove it as this task is triggered by other roles and in this case you are not responsible in those roles to remember about delegation to localhost

Reply:
No, this exact piece of code
```
    - name: Upgrade/install Helm chart
      delegate_to: localhost
      environment:
        KUBECONFIG: "{{ vault_location }}/../kubeconfig"
```
is invalid. It does nothing.

Reply:
Yes, this should be at block level

Reply:
Please look at this example:
```
---
- hosts: kubernetes_master
  tasks:
    - delegate_to: localhost
      environment:
        KUBECONFIG: somevalue
```
if you run this you'll get this error message:
```
ERROR! no module/action detected in task.

The error appears to be in '/home/asd/_ansible/prprpr/main.yml': line 4, column 7, but may
be elsewhere in the file depending on the exact syntax problem.

The offending line appears to be:

  tasks:
    - delegate_to: localhost
      ^ here
```
Please move this invalid code.

Reply:
```
ERROR! no module/action detected in task.

The error appears to be in '/home/asd/_ansible/prprpr/main.yml': line 4, column 7, but may
be elsewhere in the file depending on the exact syntax problem.

The offending line appears to be:

  tasks:
    - delegate_to: localhost
      ^ here
```
This is a result of running something similar to it. It's completely invalid.

Reply:
ill adjust to be working

Reply:
@rpudlowski93 checked and it is working fine.

Reply:
I suppose to have some misunderstanding here. 
In my opinion we can not remove these lines since we need to run the block from localhost. We just need to move the piece of code level up (tested and it works): 

```
always:
    - name: Clean up temporary Helm chart values file
      file:
        state: absent
        path: /tmp/{{ helm_chart_name }}_values.yaml
  delegate_to: localhost
  environment:
    KUBECONFIG: "{{ vault_location }}/../kubeconfig"
  block:
    # IF `disable_helm_chart`
    - when: disable_helm_chart_bool
      block:
        - name: Delete Helm chart
          shell: |
            helm delete --purge {{ helm_chart_name }}
```

Reply:
1. I'm glad it works :+1:
2. In principle it's about "removing bad lines" and "adding missing lines" :D It's not the same as "moving lines". It's not just a mistake, it's a fatal error, because the code was not tested.
3. Let's move on! :+1:😇